### PR TITLE
[16.0][MIG] marketing_crm_partner

### PR DESCRIPTION
**Pull request by yibudak**

### CODE REVIEW

**Comment by yibudak** [comment]
@pedrobaeza @CarlosRoca13 @stefan-tecnativa

**Review by flotho — APPROVED** [review]
Thanks,
Tested on runboat
Work as described.
LGTM

File: marketing_crm_partner/models/crm_lead.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class CRMLead(models.Model):
+    _inherit = "crm.lead"
+
+    @api.model
+    def _prepare_customer_values(self, name, is_company, parent_id=False):
```
Comment (chienandalu) [inline marketing_crm_partner/models/crm_lead.py]: You're changing the signature here, as the main method has this argument as optional:
```suggestion
    def _prepare_customer_values(self, name, is_company=False, parent_id=False):
```

File: marketing_crm_partner/models/crm_lead.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        """Populate marketing fields in partner."""
+        res = super()._prepare_customer_values(name, is_company, parent_id)
+        # We use self.env['utm.mixin'] for not losing possible overrides
+        # see https://github.com/odoo/odoo/blob/
+        # e5c8071484c883bf78478a39ef2120bcd8f2442d/addons/utm/models/utm.py#L51
```
Comment (chienandalu) [inline marketing_crm_partner/models/crm_lead.py]: Update the link to match the current version code: https://github.com/odoo/odoo/blob/51833e4735fb0b761d3cd2867dfd166813469e70/addons/utm/models/utm_mixin.py#L52

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks @yibudak Some remarks:

File: marketing_crm_partner/models/crm_lead.py (hunk 1 above)
Comment (yibudak) [reply marketing_crm_partner/models/crm_lead.py]: Thanks for your attention, I've fixed it up.

**Comment by CarlosRoca13** [comment]
Please fixup the 3 migration commits into this one: @yibudak
[[MIG] marketing_crm_partner: Migration to 16.0](https://github.com/OCA/crm/pull/543/commits/8a5f3c6b6b5a6c3fb0adada6b42996cfc20b87a9)

**Review by chienandalu — CHANGES_REQUESTED** [review]
@yibudak just fixup your migration commit and this would be ready to go :)

**Comment by yibudak** [comment]
@chienandalu @CarlosRoca13

**Comment by CarlosRoca13** [comment]
Tests are red :confused:

**Comment by yibudak** [comment]
> Tests are red 😕

https://github.com/OCA/crm/actions/runs/8749403166/job/24011007372?pr=543#step:8:336

Could you tell me how can I fix that? Thanks.

**Comment by chienandalu** [comment]
Maybe just rebase to the newest main branch head as the error comes from another module...

**Comment by yibudak** [comment]
@chienandalu exactly! now, all the checks are passed

**Review by chienandalu — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
You have to do a rebase, not a pull. Please amend it.

**Comment by yibudak** [comment]
@pedrobaeza Is it okay now?

**Review by pedrobaeza — APPROVED** [review]
Thanks.

/ocabot merge nobump

**Pull request merged**